### PR TITLE
GuestProxyAgent: Translate key status AuthorizationRules and validate details 

### PR DESCRIPTION
**Pull request by shahneerali**

### CODE REVIEW

File: proxy_agent/src/main.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #![allow(non_snake_case)]
-#![deny(warnings)]
+//#![deny(warnings)]
```
Comment (shahneerali) [inline proxy_agent/src/main.rs]: compile issue with authorization_rules code. functions are marked as unused even though they are being tested.

Comment (ZhidongPeng) [reply proxy_agent/src/main.rs]: mark it at your own new method instead.

File: proxy_agent_shared/src/misc_helpers.rs [hunk 2]
```diff
@@ -79,6 +79,20 @@ where
     Ok(obj)
 }
 
+pub fn json_clone<T>(obj: &T) -> std::io::Result<T>
```
Comment (ZhidongPeng) [inline proxy_agent_shared/src/misc_helpers.rs:82]: please add an easy unit test to make sure the cloned values are the same

File: proxy_agent/src/proxy/authorization_rules.rs [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+                identities: vec!["test1".to_string()],
+            }]),
+            id: "0".to_string(),
+        };
+        let rules = AuthorizationRules::from_authorization_item(authorization_item);
+        assert_eq!(rules.defaultAllowed, false);
+        assert_eq!(rules.mode, "enforce");
+        assert_eq!(rules.rules.is_some(), true);
+        
+        let url = url::Url::parse("http://localhost/test?").unwrap();
```
Comment (ZhidongPeng) [inline proxy_agent/src/proxy/authorization_rules.rs:330]: can we try use "/test?" as url

**Review by ZhidongPeng — CHANGES_REQUESTED** [review]

File: proxy_agent/src/proxy/authorization_rules.rs (hunk 3 above)
Comment (shahneerali) [reply proxy_agent/src/proxy/authorization_rules.rs:330]: no because match url::Url::parse in is_allowed requires a full path URL

**Review by ZhidongPeng — APPROVED** [review]

**Pull request merged**